tokio-trace/test_static_max_level_features/main.rs

```diff
@@ -0,0 +1,70 @@
+#[macro_use]
+extern crate tokio_trace;
+
+use std::sync::{Arc, Mutex};
+use tokio_trace::span::{Attributes, Record};
+use tokio_trace::{span, Event, Id, Level, Metadata, Subscriber};
+
+struct State {
+    last_level: Mutex<Option<Level>>,
+}
+
+struct TestSubscriber(Arc<State>);
+
+impl Subscriber for TestSubscriber {
+    fn enabled(&self, _: &Metadata) -> bool {
+        true
+    }
+
+    fn new_span(&self, _span: &Attributes) -> Id {
+        span::Id::from_u64(42)
+    }
+
+    fn record(&self, _span: &Id, _values: &Record) {}
+
+    fn record_follows_from(&self, _span: &Id, _follows: &Id) {}
+
+    fn event(&self, event: &Event) {
+        *self.0.last_level.lock().unwrap() = Some(event.metadata().level().clone());
+    }
+
+    fn enter(&self, _span: &Id) {}
+
+    fn exit(&self, _span: &Id) {}
+}
+
+fn main() {
```

Comment:
Can we change this from a main method to a test? so
```rust
#[test]
fn test_static_max_level_features() {
// ...
}
```
I think in order for `cargo test` to pick this up, we'd also want to move the file into a`tokio-trace/test_static_max_level_features/tests/` directory.